ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,308 @@
+
+__all__ = ["get_tel_ids", "get_structure", "TableLoader"]
+
+PARAMETERS_GROUP = "/dl1/event/telescope/parameters"
+IMAGES_GROUP = "/dl1/event/telescope/images"
+GEOMETRY_GROUP = "/dl2/event/subarray/geometry"
+TRIGGER_TABLE = "/dl1/event/subarray/trigger"
+SHOWER_TABLE = "/simulation/event/subarray/shower"
+TRUE_IMAGES_GROUP = "/simulation/event/telescope/images"
+
+by_id_RE = re.compile(r"tel_\d+")
+
+
+def get_tel_ids(
+    subarray: SubarrayDescription,
+    telescopes: List[Union[int, str, TelescopeDescription]]
+) -> List[int]:
+    ids = set()
+
+    for telescope in telescopes:
+        if isinstance(telescope, int):
+            ids.add(telescope)
+        ids.update(subarray.get_tel_ids_for_type(telescope))
+
+    return sorted(ids)
+
+
+def get_structure(h5file):
+
+    if PARAMETERS_GROUP in h5file:
+        g = h5file.root[PARAMETERS_GROUP]
+    elif IMAGES_GROUP in h5file:
+        g = h5file.root[IMAGES_GROUP]
+    else:
+        raise ValueError(f"No DL1 parameters data in h5file: {h5file}")
+
+    key = next(iter(g._v_children))
+    if by_id_RE.fullmatch(key):
+        return "by_id"
+    else:
+        return "by_type"
+
+
+class TableLoader(Component):
+    """ A helper class to load and join tables from a DL1 file"""
+
+    input_url = traits.Path(directory_ok=False, exists=True).tag(config=True)
+
+    load_dl1_images = traits.Bool(False).tag(config=True)
+    load_dl1_parameters = traits.Bool(True).tag(config=True)
+    load_dl2_geometry = traits.Bool(False).tag(config=True)
+    load_simulated = traits.Bool(False).tag(config=True)
+    load_true_images = traits.Bool(False).tag(config=True)
+    load_trigger = traits.Bool(True).tag(config=True)
+    load_instrument = traits.Bool(False).tag(config=True)
+
+    def __init__(self, input_url=None, **kwargs):
+        # enable using input_url as posarg
+        if input_url not in {None, traits.Undefined}:
+            kwargs["input_url"] = input_url
+        super().__init__(**kwargs)
+
+        self.subarray = SubarrayDescription.from_hdf(self.input_url)
+        self.h5file = tables.open_file(self.input_url, mode="r")
+
+        try:
+            self.structure = get_structure(self.h5file)
+        except ValueError:
+            self.structure = None
+
+        self.instrument_table = None
+        if self.load_instrument:
+            table = self.subarray.to_table()
+            optics = self.subarray.to_table(kind="optics")
+            optics["optics_index"] = np.arange(len(optics))
+            optics.remove_columns(["name", "description", "type"])
+            table = join(
+                table,
+                optics,
+                keys="optics_index",
+                # conflicts for TAB_VER, TAB_TYPE, not needed here, ignore
+                metadata_conflicts="silent",
+            )
+
+            table.remove_columns(["optics_index", "camera_index"])
+            self.instrument_table = table
+
+        self.shower_table = None
```

Comment:
Why is `shower_table` a public attribute?

Reply:
Indeed, we could move it directly in `read_subarray_event` - anything against this, @maxnoe ?

Reply:
or if it is really needed, make it `self._shower_table` since generally you don't want the users relying on it.